src/main/java/org/javarosa/xform/parse/XFormParser.java

```diff
@@ -1616,7 +1616,7 @@ public static XPathReference getAbsRef(XPathReference ref, TreeReference parentR
 
         tref = tref.parent(parentRef);
         if (tref == null) {
-            throw new XFormParseException("Binding path [" + tref + "] not allowed with parent binding of [" + parentRef + "]");
+            throw new XFormParseException("Binding path [" + ref.getReference().toString(true) + "] not allowed with parent binding of [" + parentRef + "]");

```

Comment:
Looks from the conditional block above like `ref` can be null - should we be handling that here?